### PR TITLE
[WB-5933]: Check media keys for invalid characters on windows

### PR DESCRIPTION
**Pull request by KyleGoyette**
https://wandb.atlassian.net/browse/WB-5933

Description
-----------
Check media key for forbidden chars on windows when logging media. Raise a ValueError if any are present.

Testing
-------

Adds a test to handle forbidden characters. 

Release Notes
-------------

Below, please enter user-facing release notes as one or more bullet points.
If your change is not user-visible, write `NO RELEASE NOTES` instead, with no bullet points.

------------- BEGIN RELEASE NOTES ------------------
- sanitize logged media keys on windows

------------- END RELEASE NOTES --------------------

### CODE REVIEW

**Review by raubitsj — COMMENTED** [review]
- coverage is weird
- regex and platform.system() should be compiled / cached i think

**Comment by KyleGoyette** [comment]
Coverage seems like we needed to increment the notify count to 11... Not sure when that was introduced.

regex: cached in wandb/util.py
platform just done in wandb/sdk/data_types.py, but potentially should be done somewhere more general since it's called in a variety of places

File: .codecov.yml [hunk 1]
```diff
@@ -2,9 +2,9 @@ codecov:
   require_ci_to_pass: no
   notify:
     # after_n_builds calculation
-    # 10 = 1 (lin-py36) + 4 (win-py37) + 4 (func-s_{base,tf115,tf21,tf25}-lin-py37) + 1 (pylaunch)
+    # 11 = 1 (lin-py36) + 5 (win-py37) + 4 (func-s_{base,tf115,tf21,tf25}-lin-py37) + 1 (pylaunch)
```
Comment (raubitsj) [inline .codecov.yml:5]: how is the windows count 5?  parallelism is set to 4 for windows.. it must be something else?

Comment (raubitsj) [reply .codecov.yml:5]: maybe an issue when flaky test fail, and that counts as a run so you have to have a clean set of 10 workflows run.

**Pull request merged**